addons/resource/models/utils.py

```diff
@@ -0,0 +1,41 @@
+from odoo.osv.expression import normalize_domain, NOT_OPERATOR, DOMAIN_OPERATORS
+
+def filter_domain_leaf(domain, leaf_check):
```

Comment:
In burndown chart, we could also use that method. 🙂 
(and so you can remove the function `remove_domain_leaf` in burndown chart too). 🙂